code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm

```diff
@@ -124,23 +127,30 @@
 
 	repeatable = TRUE
 
+/datum/dynamic_ruleset/latejoin/ninja/ready(var/forced = 0)
+	for(var/mob/M in candidates)
+		if(!latejoinprompt(M,src))
+			return 0
+	
+	return ..()
+
 /datum/dynamic_ruleset/latejoin/ninja/execute()
 	var/mob/M = pick(candidates)
+	M.loc = null //this is to prevent the latejoin player spawn system
```

Comment:
I feel like this should not be handled here, there needs to be a way to tell the spawn system to not spawn the player in arrivals in the first place

Reply:
the spawn system doesn't continue if its target is in nullspace - i feel like this accomplishes that as inteded